stl/inc/random

```diff
@@ -2025,6 +2027,40 @@ basic_ostream<_Elem, _Traits>& operator<<(basic_ostream<_Elem, _Traits>& _Ostr,
     return _Dist._Write(_Ostr);
 }
 
+// FUNCTION TEMPLATE _Float_upper_bound
+
+// Returns smallest _Flt such that static_cast<_Ty>(_Result) > _Val.
+// First truncate to largest _Flt <= _Val, then add ceil(ulp).
+
+template <class _Flt, class _Ty>
+_NODISCARD _Flt _Float_upper_bound(_Ty _Val) {
+    static_assert(is_unsigned_v<_Ty> && is_integral_v<_Ty> && is_floating_point_v<_Flt>,
+        "invalid template argument for _Float_upper_bound");
+    constexpr auto _Ty_digits  = numeric_limits<_Ty>::digits;
+    constexpr auto _Flt_digits = numeric_limits<_Flt>::digits;
+    using _Ty_32or64           = conditional_t<_Ty_digits <= 32, uint32_t, uint64_t>;
+
+    if _CONSTEXPR_IF (_Ty_digits <= _Flt_digits) {
+        return static_cast<_Flt>(_Val) + _Flt{1};
+    } else {
+#pragma warning(push)
+#pragma warning(disable : 4146 4293) // unary minus of unsigned, negative shift
+        constexpr auto _Mask     = _Ty(-1) << (_Ty_digits - _Flt_digits);

```

Comment:
I missed this earlier - `_Ty(-1)` is technically a functional-style cast with C cast semantics. We conventionally avoid this (because they're hard to grep for, and they're willing to do things like reinterpret pointers to/from integers, although the latter is not a danger here). Here, since you need a value-modifying conversion from `-1` to an unsigned `_Ty`, brace syntax won't work, so our style is to use a verbose-but-noticeable `static_cast`:
```suggestion
        constexpr auto _Mask     = static_cast<_Ty>(-1) << (_Ty_digits - _Flt_digits);
```